app/src/org/commcare/android/framework/CommCareActivity.java

```diff
@@ -528,14 +545,18 @@ public void onClick(DialogInterface dialog, int i) {
         };
         CharSequence buttonDisplayText = StringUtils.getStringSpannableRobust(activity, org.commcare.dalvik.R.string.ok);
         factory.setPositiveButton(buttonDisplayText, buttonListener);
-        factory.showDialog();
+        if (activity instanceof CommCareActivity) {
+            ((CommCareActivity)activity).showAlertDialog(factory);
+        } else {
+            factory.showDialog();

```

Comment:
If you change `FormNavigationUI.activity` to be a `CommCareActivity`, not `Activity`, then there won't be a need to include this conditional